### PR TITLE
Rewrite the gem to support Rails 5

### PR DESCRIPTION
**Pull request by rafaelfranca**
We found that the current version of the gem doesn't work with Rails 5 and do not raise any error.

I started this PR only to get the gem working with both Rails 4.2 and Rails 5 but the test suite was really slow and was testing the feature at the integration level (generating a Rails app, creating files, running rake tasks), so I rewrote the entire gem.

Now:

* tests are unit tests connecting to the database in the same way the Rails unit tests are doing.
* We don't have any generator.
* Works with Rails 5.
* Only including the gem already brings its behavior
* Only supports Ruby 2.2+ and Rails 4.2+

cc @camilo

### CODE REVIEW

File: lib/bigint_pk.rb [hunk 1]
```diff
@@ -1,37 +1,23 @@
...
-
-    if ca.const_defined? :PostgreSQLAdapter
-      ca::PostgreSQLAdapter::NATIVE_DATABASE_TYPES[:primary_key] = 'bigserial primary key'
-    end
-
-    if ca.const_defined? :AbstractMysqlAdapter
-      ca::AbstractMysqlAdapter::NATIVE_DATABASE_TYPES[:primary_key] = 'bigint(20) auto_increment PRIMARY KEY'
-      ca::AbstractMysqlAdapter::NATIVE_DATABASE_TYPES[:integer] = { :name => "bigint", :limit => 6 }
+  module DefaultBigintPrimaryKey
+    def primary_key(name, type = :primary_key, **options)
```
Comment (camilo) [inline lib/bigint_pk.rb:15]: why is this needed now?

File: .travis.yml [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+  - "2.3.3"
+gemfiles:
+  - Gemfile
+  - test/gemfiles/Gemfile-Rails-4.2
+env:
+  - ADAPTER=mysql2
+  - ADAPTER=postgresql
+before_script:
+  - mysql -e "create database bigint_test;"
+  - psql -c "create database bigint_test;" -U postgres
```
Comment (camilo) [inline .travis.yml]: can't this use the rake task instead?

**Review by camilo — COMMENTED** [review]
A couple questions 👍

File: lib/bigint_pk.rb (hunk 1 above)
Comment (rafaelfranca) [reply lib/bigint_pk.rb:15]: In Rails 4.2 it used the `NATIVE_DATABASE_TYPES` mapping to get the sql type to be used in the column definition. In Rails 5 this mapping is not used anymore and instead of it the ColumnDefinition object now have a DSL for all the types. So, in order to actually make this gem work with Rails 5 we need to override the primary key column builder to use bigint.

File: .travis.yml (hunk 2 above)
Comment (rafaelfranca) [reply .travis.yml]: Yes. I think it can.
